src/Runtime/Core/Maths/Series.cs

```diff
@@ -124,6 +124,104 @@ public double Evaluate(double x)
+        /// Set by the callee (coefficient generator).
+        /// </summary>
+        public double coefficient;
+        /// <summary>
+        /// state that can be used to generate the next coefficient, if success == true,
+        /// undefined otherwise
+        /// Initialized to default(InnerState) by the caller (series evaluator),
+        /// afterwards set by the callee (coefficient generator).
+        /// </summary>
+        public InnerState innerState;
+    }
+
+    /// <summary>
+    /// Provides the coefficients in the series, where the constant term is index zero.
+    /// A state is expected to be passed between the computations of consequitive coefficients
+    /// by the caller.
+    /// When the constant term (index zero) is computed, default(State) is passed to the delegate,
+    /// for a coefficient at index = n, a state obtained when computing the coefficient
+    /// at index = n - 1 is passed.
+    /// </summary>
+    /// <typeparam name="State">A type that represents the state passed between
+    /// the computations of consequitive coefficients</typeparam>
+    /// <param name="state">State passed between
+    /// the computations of consequitive coefficients</param>
+    /// <remarks>
+    /// For small coefficient generation functions, like Series.ExpMinus1Coefficient,
+    /// manually calling this delegate is about 15-30% faster that using IEnumerable.
+    /// </remarks>
+    delegate void CoefficientGenerator<State>(ref CoefficientGenerationState<State> state);
+
+    /// <summary>
+    /// Represents a mathematical power series of the form c0 + c1*x + c2*x^2 + c3*x^3 + ...
+    /// Allows stateful computation of the coefficients.
+    /// </summary>
+    struct PowerSeries<State>
+    {
+        readonly CoefficientGenerator<State> coefGenerator;
+
+        /// <summary>
+        /// Constructs a power series.
+        /// </summary>
+        /// <param name="coefGenerator">
+        /// Provides the coefficients in the series, where the constant term is index zero.
+        /// A state is expected to be passed between the computations of consequitive coefficients
+        /// by the caller.
+        /// </param>
+        public PowerSeries(CoefficientGenerator<State> coefGenerator)
+        {
+            this.coefGenerator = coefGenerator;
+        }
+
+        public double Evaluate(double x)
+        {
+            CoefficientGenerationState<State> state;
+            state.nextIndex = 0;
+            state.success = true;
+            state.coefficient = 0.0;
+            state.innerState = default(State);
+            double sum = 0.0;
+            double term = 1.0;
+            double oldSum = double.NaN;
+            coefGenerator(ref state);
+            while (state.success)
+            {
+                oldSum = sum;
+                if (state.coefficient != 0)
+                {
+                    sum += term * state.coefficient;
+                    if (sum == oldSum) break;
```

Comment:
Should use AreEqual(sum, oldSum) as done elsewhere

Reply:
Fixed.